src/routes/admin/profiles/amtProfileValidator.ts

```diff
@@ -0,0 +1,190 @@
+/*********************************************************************
+ * Copyright (c) Intel Corporation 2019
+ * SPDX-License-Identifier: Apache-2.0
+ * Author : Madhavi Losetty
+ **********************************************************************/
+import { check } from 'express-validator'
+
+export const amtProfileValidator = (): any => {
+  return [
+    check('payload.profileName')
+      .not()
+      .isEmpty()
+      .withMessage('AMT profile name is required')
+      .matches('^[a-zA-Z0-9$@$!%*#?&-_~^]+$')
+      .withMessage('AMT profile name accepts letters, numbers, special characters and no spaces'),
+    check('payload.activation')
+      .not()
+      .isEmpty()
+      .withMessage('Activation is required')
+      .isIn(['acmactivate', 'ccmactivate'])
+      .withMessage('Activation accepts either acmactivate(admin control activation) or ccmactivate(client control mode activation)')
+      .custom((value, { req }) => {
+        const pwd = req.body.payload.amtPassword
+        const randomPwd = req.body.payload.generateRandomPassword
+        if ((pwd === undefined && randomPwd === undefined)) {
```

Comment:
is null acceptable for these? In general would like to avoid comparisons for undefined since technically `undefined` can be reassigned to something. #crazyJavascript